src/components/Input/index.tsx

```diff
@@ -15,13 +14,26 @@ interface Props {
   onChange: () => void;
   onFocus: () => void;
   onKeyDown: () => void;
+  isValid: (value) => boolean;
   placeholder?: string;
   size: Size;
-  state?: State;
   value?: string;
   variablesClassName?: string;
+  validationRegex?: string;
+  validationRegexFlag?: string;
+  invalidMessage?: string;
+  limit?: number;
 }
 
+const VALID = 'valid';
+const INVALID = 'invalid';
+
+const getValidationState = (value, validationRegex, isValid) => {
+  const regex = new RegExp(validationRegex || '.*');
```

Comment:
It would be better in the default props, because it is really the default value so you wouldn't need to worry about the existence of it